src/plugins/data_source_management/public/components/create_edit_data_source_wizard/create_edit_data_source_wizard.tsx

```diff
@@ -147,20 +149,30 @@ export class CreateEditDataSourceWizard extends React.Component<
       title: [],
       description: [],
       endpoint: [],
+      credential: [],
     };
     const formErrorMessages: string[] = [];
+    /* Title validation */
     if (!this.state.dataSourceTitle) {
       validationByField.title.push('Title should not be empty');
       formErrorMessages.push('Title should not be empty');
     }
+    /* Description Validation */
     if (!this.state.dataSourceDescription) {
       validationByField.description.push('Description should not be empty');
       formErrorMessages.push('Description should not be empty');
     }
-    if (!this.state.endpoint) {
-      validationByField.endpoint.push('Endpoint should not be empty');
-      formErrorMessages.push('Endpoint should not be empty');
+    /* Endpoint Validation */
+    if (!isValidUrl(this.state.endpoint)) {
+      validationByField.endpoint.push('Endpoint is not valid');
+      formErrorMessages.push('Endpoint is not valid');
```

Comment:
nit: it might confuse the users if the URL is validate an endpoint, or the semantics is wrong. How about saying e.g. `it not valid format`?

Reply:
Wroding is upto UX. But not valid is enough. 

I feel, placeholder text or the helper text should give some example to the user about format